tools/ci-build/smithy-rs-tool-common/src/package.rs

```diff
@@ -46,3 +55,291 @@ pub enum PackageStability {
     Stable,
     Unstable,
 }
+
+/// Information required to identify a package (crate).
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub struct PackageHandle {
+    pub name: String,
+    pub version: Option<Version>,
+}
+
+impl PackageHandle {
+    pub fn new(name: impl Into<String>, version: Option<Version>) -> Self {
+        Self {
+            name: name.into(),
+            version,
+        }
+    }
+
+    /// Returns the expected owners of the crate.
+    pub fn expected_owners(&self) -> &[&str] {
+        expected_owners()
+    }
+
+    /// Returns the version number or panics
+    pub fn expect_version(&self) -> &Version {
+        if let Some(version) = &self.version {
+            version
+        } else {
+            panic!("Crate version number required for {}", self.name)
+        }
+    }
+}
+
+impl fmt::Display for PackageHandle {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self.version {
+            Some(version) => write!(f, "{}-{version}", self.name),
+            _ => f.write_str(&self.name),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub enum Publish {
+    Allowed,
+    NotAllowed,
+}
+
+/// Represents a crate (called Package since crate is a reserved word).
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub struct Package {
+    /// Package name and version information
+    pub handle: PackageHandle,
```

Comment:
If it's just name, version, etc this feels more like `meta`, perhaps it should be `PackageMeta`

Reply:
There's a lot more to meta than just name and version (authors, license, etc). This should only ever be just the name and version.